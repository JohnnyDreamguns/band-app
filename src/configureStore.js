import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

export default function configureStore(initialState = {}, history) {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createReducer(),
    initialState,
    composeWithDevTools(...enhancers)
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {};
  store.injectedSagas = {};

  return store;
}
