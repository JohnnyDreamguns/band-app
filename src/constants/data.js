export const SET_BANDS = 'band-app/Data/SET_BANDS';
export const SET_TOTAL_NUM_BANDS = 'band-app/Data/SET_TOTAL_NUM_BANDS';
export const SET_ALBUMS = 'band-app/Data/SET_ALBUMS';
export const SET_TOTAL_NUM_ALBUMS = 'band-app/Data/SET_TOTAL_NUM_ALBUMS';
