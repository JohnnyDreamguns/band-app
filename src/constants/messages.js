export const SET_ERROR = 'band-app/Messages/SET_ERROR';
export const SET_ERROR_OBJECT = 'band-app/Messages/SET_ERROR_OBJECT';
export const SET_MESSAGE = 'band-app/Messages/SET_MESSAGE';
