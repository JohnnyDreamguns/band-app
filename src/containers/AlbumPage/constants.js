export const GET_ALBUM_BY_ID = 'bands-app/AlbumPage/GET_ALBUM_BY_ID';
export const GET_BAND_BY_ID = 'bands-app/AlbumPage/GET_BAND_BY_ID';
export const SET_ALBUM_ID = 'bands-app/AlbumPage/SET_ALBUM_ID';
export const CLEAR_ALBUM_PAGE = 'bands-app/AlbumPage/CLEAR_ALBUM_PAGE';
export const SET_LOADING = 'bands-app/AlbumPage/SET_LOADING';
