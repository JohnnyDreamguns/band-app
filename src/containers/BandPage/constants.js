export const GET_BAND_BY_ID = 'bands-app/BandPage/GET_BAND_BY_ID';
export const SET_BAND_ID = 'bands-app/BandPage/SET_BAND_ID';
export const CLEAR_BAND_PAGE = 'bands-app/BandPage/CLEAR_BAND';
export const SET_ALBUM_IDS = 'bands-app/BandPage/SET_ALBUM_IDS';
export const SET_LOADING = 'bands-app/BandPage/SET_LOADING';
