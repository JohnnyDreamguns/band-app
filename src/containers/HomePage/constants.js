export const LOAD_BANDS = 'bands-app/HomePage/LOAD_BANDS';
export const SET_BAND_IDS = 'bands-app/HomePage/SET_BAND_IDS';
export const SET_PAGE_ID = 'bands-app/HomePage/SET_PAGE_ID';
export const SET_LOADING = 'bands-app/HomePage/SET_LOADING';
export const CLEAR_HOME_PAGE = 'bands-app/HomePage/CLEAR_HOME_PAGE';
