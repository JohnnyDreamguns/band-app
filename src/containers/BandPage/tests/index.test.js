import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { mount } from 'enzyme';
import { BandPage, mapStateToProps, mapDispatchToProps } from '../index';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';
import history from '../../../utils/history';
import configureStore from '../../../configureStore';
import { GET_BAND_BY_ID, CLEAR_BAND_PAGE } from '../constants';

const store = configureStore({}, history);

const renderer = new ShallowRenderer();

const mockGetBandById = jest.fn();
const mockClearBandpage = jest.fn();

const props = {
  bandId: 2,
  loading: false,
  albums: [
    {
      id: 1,
      name: 'Album Name'
    }
  ],
  band: {
    id: 1,
    name: 'Band Name',
    bio: 'Band description'
  },
  getBandById: mockGetBandById,
  clearBandpage: mockClearBandpage
};

beforeEach(() => {
  mockGetBandById.mockReset();
  mockClearBandpage.mockReset();
});

const mockState = {
  bandPage: {
    loading: true,
    bandId: 2,
    albums: [3, 4]
  },
  data: {
    bands: {
      2: {
        id: 2,
        name: 'A Band'
      }
    },
    albums: {
      3: {
        id: 3,
        name: 'An Album',
        band: 2
      },
      4: {
        id: 4,
        name: 'Another Album',
        band: 3
      }
    }
  }
};

describe('<BandPage />', () => {
  describe('component', () => {
    it('should match the snapshot', () => {
      renderer.render(<BandPage {...props} />);
      const renderedOutput = renderer.getRenderOutput();
      expect(renderedOutput).toMatchSnapshot();
    });

    it('should show loading indicator', () => {
      renderer.render(<BandPage {...props} loading={true} />);
      const renderedOutput = renderer.getRenderOutput();
      expect(renderedOutput).toMatchSnapshot();
    });

    it('should show no content if band data is missing', () => {
      renderer.render(<BandPage {...props} band={undefined} />);
      const renderedOutput = renderer.getRenderOutput();
      expect(renderedOutput).toMatchSnapshot();
    });

    it('should show no albums content if album data is missing', () => {
      renderer.render(<BandPage {...props} albums={undefined} />);
      const renderedOutput = renderer.getRenderOutput();
      expect(renderedOutput).toMatchSnapshot();
    });

    it('should call getBandById when mounted', () => {
      mount(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <BandPage {...props} />
          </ConnectedRouter>
        </Provider>
      );
      expect(mockGetBandById.mock.calls[0][0]).toBe(props.bandId);
    });

    it('should call clearBandpage when unmounted', () => {
      const wrapper = mount(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <BandPage {...props} />
          </ConnectedRouter>
        </Provider>
      );
      wrapper.unmount();
      expect(mockClearBandpage.mock.calls.length).toBe(1);
    });
  });

  describe('mapStateToProps', () => {
    it('should return the correct data', () => {
      const ownProps = {
        match: {
          params: {
            bandId: 2
          }
        }
      };
      expect(mapStateToProps(mockState, ownProps)).toEqual({
        bandId: 2,
        loading: true,
        band: { id: 2, name: 'A Band' },
        albums: [
          { id: 3, name: 'An Album', band: 2 },
          { id: 4, name: 'Another Album', band: 3 }
        ]
      });
    });
  });

  describe('mapDispatchToProps', () => {
    // it('should dispatch GET_BAND_BY_ID when getBandById is called', () => {
    //   const mockDispatch = jest.fn();
    //   const { getBandById } = mapDispatchToProps(mockDispatch);
    //   getBandById(2);

    //   expect(mockDispatch.mock.calls[0][0]).toEqual({
    //     type: GET_BAND_BY_ID,
    //     payload: 2
    //   });
    // });

    it('should dispatch CLEAR_BAND_PAGE action when clearBandpage is called', () => {
      const mockDispatch = jest.fn();
      const { clearBandpage } = mapDispatchToProps(mockDispatch);
      clearBandpage();

      expect(mockDispatch.mock.calls[0][0]).toEqual({
        type: CLEAR_BAND_PAGE
      });
    });
  });
});
