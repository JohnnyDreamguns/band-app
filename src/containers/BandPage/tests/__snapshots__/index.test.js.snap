// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BandPage /> component should match the snapshot 1`] = `
<div>
  <div>
    <h2>
      Band Name
    </h2>
    <p>
      Band description
    </p>
    <p>
      Formed: 
    </p>
    <br />
    <h3>
      Albums
    </h3>
    <p>
      <ForwardRef
        to="/album/1"
      >
        Album Name
      </ForwardRef>
    </p>
    <br />
    <p>
      Back to 
      <ForwardRef
        to="/"
      >
        Home
      </ForwardRef>
    </p>
  </div>
</div>
`;

exports[`<BandPage /> component should show loading indicator 1`] = `
<div>
  <div
    className="loader"
  />
  <div>
    <h2>
      Band Name
    </h2>
    <p>
      Band description
    </p>
    <p>
      Formed: 
    </p>
    <br />
    <h3>
      Albums
    </h3>
    <p>
      <ForwardRef
        to="/album/1"
      >
        Album Name
      </ForwardRef>
    </p>
    <br />
    <p>
      Back to 
      <ForwardRef
        to="/"
      >
        Home
      </ForwardRef>
    </p>
  </div>
</div>
`;

exports[`<BandPage /> component should show no albums content if album data is missing 1`] = `
<div>
  <div>
    <h2>
      Band Name
    </h2>
    <p>
      Band description
    </p>
    <p>
      Formed: 
    </p>
    <br />
    <h3>
      Albums
    </h3>
    <br />
    <p>
      Back to 
      <ForwardRef
        to="/"
      >
        Home
      </ForwardRef>
    </p>
  </div>
</div>
`;

exports[`<BandPage /> component should show no content if band data is missing 1`] = `<div />`;
